crates/cargo-contract/src/cmd/test.rs

```diff
@@ -32,19 +39,55 @@ use std::{
 #[derive(Debug, clap::Args)]
 #[clap(name = "test")]
 pub struct TestCommand {
+    /// Contract package to test
+    #[clap(long, short)]
+    package: Option<String>,
     /// Path to the `Cargo.toml` of the contract to test.
     #[clap(long, parse(from_os_str))]
     manifest_path: Option<PathBuf>,
+    /// Test all contract packages in the workspace.
+    #[clap(long = "--all")]
+    test_all: bool,
     #[clap(flatten)]
     verbosity: VerbosityFlags,
 }
 
 impl TestCommand {
-    pub fn exec(&self) -> Result<TestResult> {
-        let manifest_path = ManifestPath::try_from(self.manifest_path.as_ref())?;
+    pub fn exec(&self) -> Result<Vec<TestResult>> {
+        let manifest_path = match self.package.as_ref() {
+            Some(package) => {
+                let root_manifest_path =
+                    ManifestPath::try_from(self.manifest_path.as_ref())?;
+                root_manifest_path
+                    .subcontract_manifest_path(package)
+                    .context(format!(
+                        "error: package ID specification `{}` did not match any packages",
+                        package
+                    ))?
+            }
+            None => ManifestPath::try_from(self.manifest_path.as_ref())?,
```

Comment:
This pattern appears to be repeated from `build` and `check`, could it be factored out to a common method, perhaps in `ManifestPath`?